comms/comms.go

```diff
@@ -38,8 +38,8 @@ type Package interface {
 	// (agent advertisement to advertise)
 	AdvertiseAgent(defs.AgentAdvert) error
 
-	// (map of assets:hashes, lock for asset map)
-	ListenForAgent(assetProvider) error
+	// (map of assets:hashes, base asset URL)
+	ListenForAgent(assetProvider, string) error
```

Comment:
I would make this `ListenForAgent(_ assetProvider, assetURLPrefix string) error` instead of putting the description in the comments. Although it will be a bit inconsistent with the rest of the interface definition.